quinn-h3/src/proto/frame.rs

```diff
@@ -48,7 +49,11 @@ impl HttpFrame {
         let len = buf.get_var()?;
 
         if buf.remaining() < len as usize {
-            return Err(Error::Incomplete(2 + len as usize));
+            if ty == Type::DATA {
+                return Err(Error::IncompleteData);
```

Comment:
Maybe consider hoisting `return Err()` out of the `if` here?

Reply:
Done.